### PR TITLE
Allow Options with falsey keys to be selected

### LINKED ISSUE #1016: Calling state.setSelectedKey with an empty string does not update state.selectedItem
# 🐛 Bug Report 

The Picker does not display the currently selected option if the option's key is an empty string.

## 🤔 Expected Behavior

The trigger button should display the selected item.

## 😯 Current Behavior

The trigger button does not display the rendered selected item. However, when you open the Picker dialog, you see that the item is correctly selected.

image

## 💁 Possible Solution

`state.selectedItem` should update when `state.setSelectionKey` is called with an empty string

## 🔦 Context

I was attempting to emulate default browser behavior where it would select the first option if none of the options are currently selected. In addition, I was attempting to provide a different placeholder value. In regular HTML, the convention is to provide a "fake" option as the first item whose value is an empty string.

```jsx
<Picker label="Foo">
  <Item key="">Select a color</Item>
  <Item key="yellow">Yellow</Item>
  <Item key="blue">Blue</Item>
</Picker>
```

## 💻 Code Sample

Here's a PR illustrating the bug https://github.com/adobe/react-spectrum/pull/1015

Although the PR focuses on selectedKey/defaultSelected key, you can trigger this bug with normal interactions as long as the Item key is an empty string.

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-spectrum   | 3.2.0
| Browser          | Chrome 84.0.4147.135
| Operating System | MacOS Catalina (10.15.6)

## 🧢 Your Company/Team

Polarwind, Envoy's component library

### PR DESCRIPTION
**Pull request by kamal**
This PR allows Picker options which have falsey keys (except `null`) to be selected. The use case where a user might encounter this is when a Picker contains an Option with `key=""` as would be the convention in regular HTML when creating a placeholder value for a select input.

Closes #1016

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [x] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [x] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

1. Construct a Picker with an option containing an empty key
    ```jsx
    <Picker label="Color">
      <Option key="">None</Option>
      <Option key="red">Red</Option>
      <Option key="blue">Blue</Option>
      <Option key="green">Green</Option>
    </Picker>
    ```
2. Select the "None" option
3. Picker should now show "None" in the Picker button

### CODE REVIEW

**Comment by snowystinger** [comment]
Thanks for updating this!

**Review by dannify — APPROVED** [review]

**Comment by kamal** [comment]
Since it's been approved, but now not up to date with master, do you rebase and squash just before you merge or do you expect me to rebase?

**Review by devongovett — APPROVED** [review]

**Comment by devongovett** [comment]
I think we were just waiting for the last release to go out. Thanks again!

**Pull request merged**
